src/components/UI/Modal/LoginModal/LoginModal.js

```diff
@@ -0,0 +1,130 @@
+import {
+  ChainInfo,
+  LoginErrorType,
+  NetworkType,
+  WalletErrorType,
+  WalletStatus
+} from '@starkware-industries/commons-js-enums';
+import PropTypes from 'prop-types';
+import React, {useEffect, useState} from 'react';
+
+import {
+  useEnvs,
+  useLoginTracking,
+  useLoginTranslation,
+  useWalletHandlerProvider,
+  useDidMountEffect
+} from '../../../../hooks';
+import {useLogin} from '../../../../providers/AppProvider';
+import {useHideModal} from '../../../../providers/ModalProvider';
+import {useLoginWallet, useWalletsStatus} from '../../../../providers/WalletsProvider';
+import {evaluate, isChrome, isFirefox} from '../../../../utils';
+import {MultiChoiceMenu} from '../../index';
+import styles from './LoginModal.module.scss';
+
+const AUTO_CONNECT_TIMEOUT_DURATION = 100;
+
+const LoginModal = ({networkName}) => {
+  const {titleTxt, unsupportedBrowserTxt, unsupportedChainIdTxt} = useLoginTranslation();
+  const [trackLoginScreen, trackWalletClick, trackLoginError] = useLoginTracking();
+  const {AUTO_CONNECT, SUPPORTED_L1_CHAIN_ID} = useEnvs();
+  const [error, setError] = useState(null);
+  const {statusL1, statusL2} = useWalletsStatus();
+  const [network, setNetwork] = useState(networkName || NetworkType.L1);
+  const {walletError, walletStatus, connectWallet} = useLoginWallet(network);
+  const walletHandlers = useWalletHandlerProvider(network);
+  const {isLoggedIn} = useLogin();
+  const hideModal = useHideModal();
+
+  useEffect(() => {
+    isLoggedIn && hideModal();
+  }, [isLoggedIn]);
+
+  useEffect(() => {
```

Comment:
Shouldn't be handled on the modal but on the main menu screen.

Reply:
Can you explain why? 
The `hideModal` function is a global function for all the modals. If this function will be on the main menu screen, couldn't it affect other modals?